rdflib/query.py

```diff
@@ -57,8 +57,8 @@ def __init__(self, graph: "Graph"):
     def query(  # type: ignore[empty-body]
         self,
         strOrQuery: Union[str, "Query"],  # noqa: N803
-        initBindings: Mapping["str", "Identifier"] = {},  # noqa: N803
-        initNs: Mapping[str, Any] = {},  # noqa: N803
+        initBindings: Optional[Mapping["str", "Identifier"]] = None,  # noqa: N803
+        initNs: Optional[Mapping[str, Any]] = None,  # noqa: N803
```

Comment:
While the current function signature is problematic due to its use of mutable data structures in argument defaults, One problem with changing this is, plugins that implement this as it stands could break if they now receive valid values according to the new function signature, as they don't expect `None` as a value.

We need to figure out a way to upgrade our plugin interfaces, I think the right approach is to create ProcessorV2, but we need to also consider what other changes we want to introduce before that.

I think it is best to change this back to what it was, though the rest of the PR seems good and quite helpful.

```suggestion
        initBindings: Mapping["str", "Identifier"] = {},  # noqa: N803
        initNs: Mapping[str, Any] = {},  # noqa: N803
```

Reply:
I see how this could be an issue. I will switch the defaults to retain the empty dictionary

Reply:
Addressed in 259c10db and 0ad0cffe

Reply:
Done now, please have a look and let me know if things look okay, but essentially I just reverted https://github.com/RDFLib/rdflib/commit/259c10db10e5ead9f48da007cf30d28af1c9b9a9 and then changed `rdflib/query.py` back to what is in the main branch.